server/src/main/java/org/opensearch/cluster/metadata/View.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.metadata;
+
+import org.opensearch.cluster.AbstractDiffable;
+import org.opensearch.cluster.Diff;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.core.ParseField;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ConstructingObjectParser;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+/** View of data in OpenSearch indices */
+@ExperimentalApi
+public class View extends AbstractDiffable<View> implements ToXContentObject {
+
+    private final String name;
+    private final String description;
+    private final long createdAt;
+    private final long modifiedAt;
+    private final List<Target> targets;
```

Comment:
I am wondering if `List<Target>` is a good fit here, we definitely don't want duplicate targets here, and probably having consistent order (to eliminate unnecessary modifications)

Reply:
Changing this over to sorted set - might as well do this on the initial revision